### ISSUE TITLE
Support the `file` attribute under `<testcase>`

### ISSUE THREAD

**Issue by wcedmisten-reify** [lambdaisland/kaocha-junit-xml#17]
The CircleCI docs expect a `file` attribute under the `<testcase>` or `<testsuite>` tags to lookup test timings by filename.

https://circleci.com/docs/use-the-circleci-cli-to-split-tests/#junit-xml-reports

Although I can't find this mentioned anywhere in the Junit schemas linked in the README, would you consider supporting this in your plugin?

**Comment by alysbrooks** [comment]
Hi @wcedmisten-reify, thanks for suggesting this. 

So implementing this would allow clicking on a file name to see all the timing for that file? Am I understanding you (and the CircleCI docs) correctly? That definitely seems useful.

**Comment by lambduhh** [comment]
Meowdy @wcedmisten-reify 

Thanks for bringing this to our attention! I reviewed during triage and introducing the file attribute in our JUnit XML outputs could significantly benefit CircleCI users. This enhancement not only aims to speed up test runs but becomes high priority since it could also **lead to cost savings by improving test parallelization efficiency.**

Rationale: CircleCI calculates charges based on actual test run times, not the aggregate job time. Efficient parallelization means less idle time and, consequently, reduced costs for users with extensive test suites. This feature would enable CircleCI to allocate tests based on precise file execution times, optimizing resource use and decreasing overall test duration.

In short, by making this update, we're looking at **faster CI builds and potential savings for our users on CircleCI.** It’s a strategic improvement that aligns with our commitment to enhance user experience and operational efficiency.

I'm excited about the potential here and believe it deserves our immediate attention. Changing status to "candidate" and updating to "high" priority. 

References:
https://circleci.com/docs/parallelism-faster-jobs/

**Comment by wcedmisten-reify** [comment]
It looks like there is [already a flag](https://github.com/lambdaisland/kaocha-junit-xml/pull/18) that addresses this issue: `--junit-xml-add-location-metadata`

With it, I was able to get CircleCI test splitting working as expected (although I did need a hacky `sed` command to convert the absolute paths to relative ones in order for CircleCI to recognize the timing data).

Is there a reason this flag isn't enabled by default? I would propose enabling it unless there are serious objections to that behavior

**Comment by nnichols** [comment]
I updated the linked [PR](https://github.com/lambdaisland/kaocha-junit-xml/pull/18) to add a new flag to switch between relative/absolute pathing: `--junit-xml-use-relative-path-in-location`. (Biasing towards backwards compatibility in case different CI Platforms have different pathing expectations).

I don't have a dog in the race for enabling `--junit-xml-add-location-metadata` by default or not. I'll happily defer to the maintainers on that one
